tasks/ssh.yml

```diff
@@ -12,13 +12,51 @@
     src: "{{ kdump_sshkey }}.pub"
   register: keydata
 
-- name: Add key to authorized keys
-  authorized_key:
-    user: "{{ kdump_ssh_user }}"
-    key: "{{ keydata.content|b64decode }}"
+- name: Get userinfo for "{{ kdump_ssh_user }}"
+  user:
+    name: "{{ kdump_ssh_user }}"
     state: present
+  register: __kdump_ssh_user_info
   delegate_to: "{{ kdump_ssh_server }}"
 
+- name: Get the authorized_keys file for the user
+  stat:
+    path: "{{ __kdump_ssh_user_info.home ~ '/.ssh/authorized_keys' }}"
+  register: __kdump_authorized_keys_file
+  delegate_to: "{{ kdump_ssh_server }}"
+
+- name: Get the authorized_keys contents, if any
+  slurp:
+    src: "{{ __kdump_authorized_keys_file.stat.path }}"
+  delegate_to: "{{ kdump_ssh_server }}"
+  register: __kdump_authorized_keys
+  when:
+    - __kdump_authorized_keys_file.stat.isreg is defined
+    - __kdump_authorized_keys_file.stat.isreg
+
+- name: Write new authorized_keys if needed
+  vars:
+    __kdump_authorized_keys_lines: "{{
+      (__kdump_authorized_keys.content | b64decode).split('\n') |
+      reject('match', '^$') | list
+      if __kdump_authorized_keys is defined else []
+    }}"
+    __kdump_authorized_keys_path: "{{
+      __kdump_authorized_keys_file.stat.path |
+      d(__kdump_ssh_user_info.home ~ '/.ssh/authorized_keys')
```

Comment:
In a task above, `"{{ __kdump_ssh_user_info.home ~ '/.ssh/authorized_keys' }}"` is registered into `__kdump_authorized_keys_file`. Why do you provide the default here?

Reply:
> In a task above, `"{{ __kdump_ssh_user_info.home ~ '/.ssh/authorized_keys' }}"` is registered into `__kdump_authorized_keys_file`.

Only if the file exists.  If the file doesn't exist, then `__kdump_authorized_keys_file.stat.path` will be undefined.

> Why do you provide the default value with `d(..)` here?